allauth/account/forms.py

```diff
@@ -598,18 +598,14 @@ def clean_email(self):
             raise get_adapter().validation_error("unknown_email")
         return self.cleaned_data["email"]
 
-    def save(self, request, **kwargs):
+    def save(self, request, **kwargs) -> str:
         email = self.cleaned_data["email"]
         if not self.users:
             flows.signup.send_unknown_account_mail(request, email)
-        else:
-            self._send_password_reset_mail(request, email, self.users, **kwargs)
-        return email
 
-    def _send_password_reset_mail(self, request, email, users, **kwargs):
+        adapter: DefaultAccountAdapter = get_adapter()
```

Comment:
I think it would be good to either abort the method inside the `if` above by returning `return email`, or, to reintroduce the `else`. As is now, both cases are executed (the if and the else part), and while that works because `users` is empty it did trip me over while reviewing.

Reply:
Ah, good catch though!